cachi2/core/resolver.py

```diff
@@ -14,13 +14,19 @@
     "pip": pip.fetch_pip_source,
 }
 
+_dev_package_managers: dict[PackageManagerType, Handler] = {
+    "yarn": yarn.fetch_yarn_source,
+}
 
 supported_package_managers = list(_package_managers)
 
 
 def resolve_packages(request: Request) -> RequestOutput:
     """Run all requested package managers, return their combined output."""
+    global _package_managers
```

Comment:
Unless I'm misunderstanding the PEP584 docs, it seemed to me that this _should_ work without declaring _package_managers as a global, since the "|=" operator supposedly modifies the dict in-place, but it doesn't :thinking: 

After I went down a rabbit hole on that, I wonder if we can sidestep this entirely and just use local variables instead?

Reply:
`_package_managers` has to be declared `global` in the local scope so that it can be modified *at all*. Regardless, I'm happier with just using a local variable. `global` makes the code more concise, but it's not used very often at all, which can lead to confusion for people who've never seen it.